components/locale_canonicalizer/src/provider.rs

```diff
@@ -82,30 +85,29 @@ pub struct AliasesV1<'data> {
 /// defined for the likely subtags maximize algorithm.
 ///
 /// For efficiency, only the relevant part of the LanguageIdentifier is stored
-/// for searching. E.g., the `language_script` field is used to store rules for
-/// `LanguageIdentifier` that contain a language and a script, but not a region.
-/// This also allows for space savings by using a `TinyStr4` rather than a full
-/// `LanguageIdentifier`.
-pub struct LikelySubtagsV1 {
+/// for searching and replacing. E.g., the `language_script` field is used to store
+/// rules for `LanguageIdentifier`s that contain a language and a script, but not a
+/// region.
+#[yoke(prove_covariance_manually)]
+pub struct LikelySubtagsV1<'data> {
     /// Language and script.
-    #[zerofrom(clone)]
-    pub language_script: LiteMap<(TinyAsciiStr<3>, TinyAsciiStr<4>), LanguageIdentifier>,
+    #[cfg_attr(feature = "serialize", serde(borrow))]
+    pub language_script: ZeroMap<'data, (Language, Script), Region>,
```

Comment:
Discussion: For lookup (the left side of the map), we can use "raw" types, because we don't need to validate them. We only need to validate things if they are used for retrieval (the right side of the map).

Reply:
Good point. We should also update the value types in `AliasesV1` to be validated.